doc/developers/contributing.rst

```diff
@@ -155,12 +155,23 @@ If any of the above seems like magic to you, then look up the `Git documentation
 <http://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html>`_ on the
 web.
 
-In particular, if some conflicts arise between your branch and the master
-branch, you will need to `rebase your branch on master
-<http://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#rebasing-on-master>`_.
-Please avoid merging master branch into yours. If you did it anyway, you can fix
-it following `this example
-<https://github.com/scikit-learn/scikit-learn/pull/7111#issuecomment-249175383>`_.
+If some conflicts arise between your branch and the ``master`` branch, you need
+to merge ``master`` into your branch with the explicit ``--no-ff`` flag. The
+command will be::
+
+  $ git merge --no-ff master
```

Comment:
Why do you require the --no-ff flag ?

If there is a conflict, there will be no fast forward possible. Actually, even if there is no commit but some change on both branch, a fast forward will not be possible.